triton-vm/src/vm.rs

```diff
@@ -971,40 +971,34 @@ pub mod triton_vm_tests {
             let ext_processor_table =
                 processor_table.extend(&challenges.processor_table_challenges, interpolant_degree);
 
-            for (row_idx, (row, next_row)) in ext_processor_table
+            for (row_idx, (current_row, next_row)) in ext_processor_table
                 .data()
                 .iter()
                 .tuple_windows()
                 .enumerate()
             {
-                let evaluation_point = vec![row.clone(), next_row.clone()].concat();
-
                 for (tc_idx, tc_evaluation_result) in ext_processor_table
-                    .evaluate_transition_constraints(&evaluation_point, &challenges)
+                    .evaluate_transition_constraints(current_row, next_row, &challenges)
                     .iter()
                     .enumerate()
                 {
                     if !tc_evaluation_result.is_zero() {
+                        let ci = current_row[ProcessorBaseTableColumn::CI as usize].coefficients[0]
+                            .value();
```

Comment:
I'm aware you only extracted this from further down. However, both `XFieldElement` and `BFieldElement` have a pretty decent `Display` implementation by now – simply using `format!("{}", …)` probably yields a very readable result.

Reply:
Ah -- we could use `Display` instead of `Debug` here.

The point of declaring `ci` here is that it's a rather long line and repeats.